lib/ansible/modules/cloud/amazon/aws_acm.py

```diff
@@ -0,0 +1,396 @@
+    region: ap-southeast-2
+
+- name: delete a certificate with a particular ARN
+  aws_acm:
+    certificate_arn: "arn:aws:acm:ap-southeast-2:123456789012:certificate/01234567-abcd-abcd-abcd-012345678901"
+    state: absent
+    region: ap-southeast-2
+
+- name: delete all certificates with a particular domain name
+  aws_acm:
+    domain_name: acm.ansible.com
+    state: absent
+    region: ap-southeast-2
+
+'''
+
+RETURN = '''
+certificate:
+  description: Information about the certificate which was uploaded
+  type: complex
+  returned: when C(state) is C(present)
+  contains:
+    arn:
+      description: The ARN of the certificate in ACM
+      type: str
+      returned: when C(state) is C(present)
+      sample: "arn:aws:acm:ap-southeast-2:123456789012:certificate/01234567-abcd-abcd-abcd-012345678901"
+    domain_name:
+      description: The domain name encoded within the public certificate
+      type: str
+      returned: when C(state) is C(present)
+      sample: acm.ansible.com
+arns:
+  description: A list of the ARNs of the certificates in ACM which were deleted
+  type: list
+  returned: when C(state) is C(absent)
+  sample:
+   - "arn:aws:acm:ap-southeast-2:123456789012:certificate/01234567-abcd-abcd-abcd-012345678901"
+'''
+
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info
+from ansible.module_utils.aws.acm import ACMServiceManager
+from ansible.module_utils.crypto import get_fingerprint_from_pem_cert as get_fingerprint
+from ansible.module_utils._text import to_bytes, to_text
+from ssl import PEM_cert_to_DER_cert
+
+import re  # regex library
+
+
+# Takes in two text arguments
+# Each a PEM encoded certificate
+# Or a chain of PEM encoded certificates
+# May include some lines between each chain in the cert, e.g. "Subject: ..."
+# Returns True iff the chains/certs are functionally identical (including chain order)
+def chain_compare(a, b):
+
+    # Use regex to split up a chain or single cert into an array of base64 encoded data
+    # Using "-----BEGIN CERTIFICATE-----" and "----END CERTIFICATE----"
+    # Noting that some chains have non-pem data in between each cert
+    expr = re.compile(r"-+BEGIN\s+CERTIFICATE-+([a-zA-Z0-9\+\/=\s]+)-+END\s+CERTIFICATE-+")
+    chain_a_pem = pem_chain_split(a)
+    chain_b_pem = pem_chain_split(b)
+
+    if len(chain_a_pem) != len(chain_b_pem):
+        return(False)
+
+    # Chain length is the same
+    for (ca, cb) in zip(chain_a_pem, chain_b_pem):
+        der_a = PEM_body_to_DER(ca)
+        der_b = PEM_body_to_DER(cb)
+        if der_a != der_b:
+            return(False)
+
+    return(True)
+
+
+# Takes in PEM encoded data with no headers
+# returns equivilent DER
+def PEM_body_to_DER(pem):
+    prefix = "-----BEGIN CERTIFICATE-----"
+    suffix = "-----END CERTIFICATE-----"
+    pem_full = '\n'.join([prefix, pem, suffix])
+    return(PEM_cert_to_DER_cert(pem_full))
```

Comment:
You can also simply use `base64.b64decode()` instead of first re-adding the headers and then calling `PEM_cert_to_DER_cert()`. Also, you should check whether `PEM_cert_to_DER_cert()` (or `base64.b64decode()` if you use that) raises an exception, and react to it if necessary.

(BTW, this is how `PEM_cert_to_DER_cert()` can be implemented: https://github.com/python/cpython/blob/master/Lib/ssl.py#L1456-L1467)

Reply:
With Python 3.6, I've seen `ValueError` (incorrect header) and `binascii.Error` (bad Base64).

Reply:
I just changed it to `b64decode`.